source/grid/manifold.cc

```diff
@@ -101,9 +101,15 @@ Manifold<dim, spacedim>::get_new_point(
         weight = w / (weights[permutation[i]] + w);
 
       if (std::abs(weight) > 1e-14)
-        p = get_intermediate_point(p,
-                                   surrounding_points[permutation[i]],
-                                   1.0 - weight);
+        {
+          p = get_intermediate_point(p,
+                                     surrounding_points[permutation[i]],
+                                     1.0 - weight);
+        }
+      else
+        {
+          p = surrounding_points[permutation[i]];
+        }
```

Comment:
Actually, this optimization should not be necessary at all. In `get_intermediate_point` we do the same check with `tolerance=1.e-10`.

Reply:
True, unless we derive the `Manifold` class and only implement `get_intermediate_point` as suggested by the documentation. Either way, this optimization doesn't change much. Let me know if you'd rather take it out.

Reply:
It's fine.